### PR TITLE
PubbyStation Engine Overhaul

### PR DESCRIPTION
**Pull request by SergeiKoralev**
## About The Pull Request

By the request of Crossedfall I made a TEG engine for pubby station. This is the result of that request.

## Why It's Good For The Game

Radiation engines are easy. If you want real engineers you take a TEG SM reactor.

## Changelog
:cl:
add: Overhauled engineering compartment with new and improved TEG setup.
add: A new, fancy SM chamber monitoring console so you can see exactly how fucked your gas mix is without a SM monitor. (Useful for if the SM ever gets dragged away somehow.)
del: The unrobust Radiation Collector engine.
tweak: tweaked engineering gameplay slightly, and probably a few engineering mains' nerves.
/:cl:

Below is a proof of functionality test. This setup has not been changed from the round-start configuration.
9c7723fc737fb25672dc372c5b718c37

### CODE REVIEW

**Comment by ike709** [comment]
oof I see we need to port Oracle's TEG sprites

**Comment by SergeiKoralev** [comment]
I mean that's not my problem.

**Comment by ike709** [comment]
Don't recall saying it was.

**Comment by VictorVow** [comment]
Change Engineering doors access so Atmos Techs can enter too. It was already needed because of the protolathe being in Engi but even more necessary with a TEG.

**Comment by SergeiKoralev** [comment]
Personally I'd rather go into the high capacity access system and just give atmos techs basic engineering access even on high pop... but that's not part of this PR.

**Comment by Crossedfall** [comment]
Test merging onto Sage

**Comment by Isy232** [comment]
The SM air alarm starts off as locked, meaning engineers cant do anything to it to start the engine.

Other than that I like

**Comment by Isy232** [comment]
image
TEG does not work with base setup, you need to get more plasma for the cold loop in order for it it function.
Engineers and atmos needs mutual access in order for this to function.
Atmos to SM loop could also use a connection to the cold loop.

**Comment by Isy232** [comment]
Could you add visual indications like arrows on the floor to indicate how cold and warm gas has to flow?

**Comment by SergeiKoralev** [comment]
> image
> TEG does not work with base setup, you need to get more plasma for the cold loop in order for it it function.
> Engineers and atmos needs mutual access in order for this to function.
> Atmos to SM loop could also use a connection to the cold loop.

TEG does work with base setup. Stop changing the setup and you'll see that it works.
TEG's for dummies
-put gasses in hot loop
-put gasses in cold loop
-filter O2, Plasma, CO2 with scrubbers on expanded or normal, and filter from hot loop after passing through TEG
-turn on emitters
-profit.

Reference: Picture in PR description.

**Comment by SergeiKoralev** [comment]
> Could you add visual indications like arrows on the floor to indicate how cold and warm gas has to flow?

Arrows for flow direction are issued as starting pumps.

**Comment by Crossedfall** [comment]
> 
> 
> > Could you add visual indications like arrows on the floor to indicate how cold and warm gas has to flow?
> 
> Arrows for flow direction are issued as starting pumps.

Having actual arrows under the TEG might not be a terrible idea though. Just to give that extra bit of information. Since it's easy to forget.

I'd imagine we could add them under the two pipes going into the bottom of the TEG, indicating the expected pressure flow.

**Comment by SergeiKoralev** [comment]
I'll put in some pixle shifted arrows next to the pipes and see if travis complains about that. I agree, it is actually a decent idea.

**Comment by Isy232** [comment]
https://cdn.discordapp.com/attachments/419192460909215756/702303450880409691/unknown.png

**Comment by SergeiKoralev** [comment]
> https://cdn.discordapp.com/attachments/419192460909215756/702303450880409691/unknown.png

that's uh... an interesting problem... also not mine. 
image

**Comment by Isy232** [comment]
@Vasily2013 Mine 
image

**Comment by Isy232** [comment]
image

**Comment by Isy232** [comment]
Uses pipes to generate an extrmely got plasma fire, only restriction is that it cannot run continuously as the plasma fire pipes will eventually be drained, then close the valve and refill

**Comment by Isy232** [comment]
image

**Comment by Isy232** [comment]
Its not stable at the extremely high productions, however interestingly enough it generates more power seemingly when I open it for short bursts rather than all at once.

I like bragging, but this PR good.

**Comment by Isy232** [comment]
Note that the setup I am using does not involve the Sm at all except for the cooling loop having a heat exchanger to SM loop so SM does not go kablooey.

**Comment by Isy232** [comment]
While your at is, please move the air srcubber and vent pipes far out of the way of the SM/TEG setup, they are such a pain and I accidentally connected my plasma loop to them once....

**Comment by SergeiKoralev** [comment]
> While your at is, please move the air srcubber and vent pipes far out of the way of the SM/TEG setup, they are such a pain and I accidentally connected my plasma loop to them once....

Think of it as just another thing to consider when setting up chad fusion pipe burn TEGs. (Keep the t-ray scanner on and in your pocket. It helps.)

**Comment by Ghilker** [comment]
so you made a TEG SM without changing the SM code to make it work with heat? why

**Comment by ike709** [comment]
@Ghilker TEG SM is viable already, see: FTL13 using it as their engine setup throughout its entire existence. Pretty sure Oracle used it too.

Any issues with it are a mapping problem or an engineer incompetence problem.

**Comment by Ghilker** [comment]
does this SM use tg code? are the gases interaction the same?

**Comment by Cornarias** [comment]
Hello, big ol' atmos boi over here. Plenty hundred hours in atmos and engineering. I'm pretty bloody knowledge when it comes to the SM and gasses in general.

This change is, frankly, not good. In so many ways. Lemme break it down:


1. TEGs are boring and overpowered.

Plain and simply. The best way to use this SM you've made is to ignore most of it. Just have the SM flair itself up on repeat then go down. Easy setup to do, tonnes of power. Minimal effort.

Or, better yet, have atmos send you a burn over. Even at low amounts and pressures, it'll put out some absurd power. Totally ignoring the SM in its entirety. It's also far, far harder to sabotage a TEG in an interesting a meaningful way. And if you sabotage the supermatter, well, who cares?

Because you can just use a heater or two, and bypass any of what I just said.

TEG SMs are interesting because of the customization to the roundstart SM, not because of the TEG.


2. N2O... In a setup where you want to heat the SM?

N2O decays at high temperatures. Yes it reinforces heat resistance, but in a setup where you supposedly want people balancing heat in higher levels... It's just not a good decision. Because at the temps it reinforces at, it'll decay. It's better to just add juuuust enough N2 that the heat penalty isn't too bad and let it go from there. Or add CO2, harvest pluoxium, and dump that in, then light her up. Because pluoxium's downside of radiation dampening is not a downside, but a straight upgrade!


3. Less customizability

Currently the SM has a large number of gasses that focus around balancing heat, power level, radiation, and gasses. Tritium gives exponentially high levels of power, but is very hard to contol at high volumes even for seasoned engies. BZ is weaker and safer, but also puts out neutrons. Pluoxium actually has a downside. CO2 is a probably and overpowered gas considering how easy it is for the huge amount of power it generates, but also allows for interesting 'disco' / 'baby tesla' setups which are very dangerous but very fun, interesting, exciting, and beneficial. ETC.

This is because a lot of the N2 is focused around the balancing act of the gas heat penalty, gas SHCs, and waste such as rads produced.

This change entirely removes the radiation aspect, caring only about heat, which is hilariously easy to achieve and shuts down a lot of interesting engineering customizations. 1x1, 1x3, large area, and other similar chamber modifications are now pointless. Managing gas mixes is a breeze. And the basic setup is as easy as ever.


4. This is just a bad version of what Aurora has.

Aurora's TEG SM actually has significant modifications to make this work. And I'll be honest, it's still not that fun. Or engaging. But Aurora is HRP, mechanics traditionally aren't the focus, so it's fine. This is all the issues of that, without what few upsides it has.


5. I don't understand why you think this is any more difficult than the standard SM when it isn't. At all. Even if we *were* required to use the SM to get the heat needed to power the TEG (which is not the case so long as atmos exists), I don't see the benefits of this engine over the standard SM. There's straight up less to do. Even as far as min-maxing power is concerned.


Basically, if you want a TEG SM, this ain't it, chief. Both the SM and the TEG need major overhauls for something like this to work. The TEG needs nerfs and fixes and actual danger, the SM needs to be prepped and reworked for being a heat-based system (with gasses being rebalanced around that), and it needs to be done in a way that keeps the balancing act of the current SM.

As it stands, this seems like change for the sake of change. Worse still, change for a more boring and simplistic alternative.

EDIT: For Ghilker, unless I've missed a pretty huge PR, I have the gas defines and interactions up on the screen from this github right now. Gas works as you'd expect for the most part.

**Comment by Ghilker** [comment]
then this PR is stupid and shouldnt exist at all

**Comment by Isy232** [comment]
>And if you sabotage the supermatter, well, who cares?

SM go boom.

Also this engine works as intended, its good. The TEG is a bit OP if you use a big plasma burn mix, but everything else is fine.

You can easily run this engine just with a bit of engineering know-how and its nice to see something different in rotation.

For all who want to do fun SM stuff you still can, maybe with a bit more work. Nothing is stopping you from constructing rad collectors.

**Comment by Crossedfall** [comment]
> This change entirely removes the radiation aspect

Good. Moving away from radiation is the definite future. This is inevitably part of that.

**Comment by Cornarias** [comment]
> 
> 
> > And if you sabotage the supermatter, well, who cares?
> 
> SM go boom.
> 
> Also this engine works as intended, its good. The TEG is a bit OP if you use a big plasma burn mix, but everything else is fine.
> 
> You can easily run this engine just with a bit of engineering know-how and its nice to see something different in rotation.
> 
> For all who want to do fun SM stuff you still can, maybe with a bit more work. Nothing is stopping you from constructing rad collectors.

SM go boom? Not at all. Just dump in some cooled N2 and remove it from the loop. It's unnecessary. It's also much easier to repair.

As for having to then construct rad collectors, I'm not sure why going through the extra steps to _then_ be allowed to return to the original and far more customizable should be required.

**Comment by ike709** [comment]
@Cornarias Radiation as a mechanic is on its way out due to major performance problems and other issues. It has never been a sound mechanic, even the guy that created radiation code acknowledges its faults and wants it gone. BYOND simply can't handle it; it can be semi-reliably used to crash the server.

It's TEG SM or back to Lord Singuloth or Memeball.

**Comment by Cornarias** [comment]
@ike709 

Fair. But is this radiation due to abuse such as 'rad boosting' with BRPEDs and such, or its natural state?

And Singulo also generates rads, unless that's intended to change?

Is removing radiation and changing the SM to function in a similar manner out of the question? As losing the SM is a hell of a shame, considering how interesting and enjoyable it is for the creative engineer. Whereas SM TEGs are far less so. Ideally the SM wouldn't be part of the equation, and it would focus on the TEG in a similar manner to Goon's combustion engine (Similar, obviously. Goonmos is out of the question)

**Comment by ike709** [comment]
Natural state. 

With regards to singuloth/non-TEG SM, the only option would be the pre-radiation method of having rad collectors simply check that the engine exists. And that's no fun.

At least TEG SM has the *potential* for depth if engineers want to dive in and build a burn chamber and other wacky shenanigans, rather than cheese it.

**Comment by ike709** [comment]
Just to give you an idea, radiation runs much worse than atmos despite affecting significantly less of the game under normal circumstances.

**Comment by Cornarias** [comment]
Oh don't worry, I know that radiation runs like hot arse. It's a huge issue, especially in its current state. My biggest concern is why the SM can't be worked around it. It's a bit snowflakey, yes, but so is the singulo without radiation.

Something like pulses of energy unique to the SM. It keeps what makes it a fun and enjoyable engine while still allowing for far, far better performance.

Also, having two chats at once on this topic happening in essentially the same order is trippy.

EDIT: I should clarify that I was unaware that some of the intent with this PR is processing power of rads originally. It makes a lot more sense coming from that angle, but I still feel there are other solutions.

**Comment by SergeiKoralev** [comment]
In rebuttal to the above.

Ghilker, Bee uses a variant of TG code. Yes. As for "Making it work with heat" It does already work with heat, see example picture in PR. 

Cornarias, I'll offer a point by point rebuttal.
1: TEG's are boring - Well, in my experience radiation engines are just as boring. You try to keep the crystal as cold as possible with as high an EER as possible, and overall when playing with radiation engines I find myself bored out of my mind trying to find something else to do, rather than experimenting with tweaks to my scrubber settings, gas mix, filters, or pipe net. Every engine has a boring variant, and every engine has ways to make it more interesting. The engine is what you make of it.

2: N2O in a heat-based setup - Yes, the base setup suggests that you use N2O. This is because unlike an atmosian elder such as you and I, many engineers are not enlightened in the ways of atmos. The default setup provides something that, if set up in a reasonable manor, won't explode randomly, yet still provides sufficient power to keep the lights on.

3: Less customization - By any chance were you an FTL13 veteran? I saw more versions of TEG setups in FTL13 than I've ever seen radiation setups. Rarely do you see an engineer experimenting with changing the pipe layout of a radiation net in my experience. Radiation engineers do not yet truly understand the greatness that you can do with a shard and an RPD. Wait and watch as all the different variants come out of the woodwork. 

4: See again point 1, the engine is what you make of it.

5: I literally watched the first round of this PR being test merged. The engineers on the shift almost blew it up because they didn't understand how a TEG's cooling system works. The round after I watched them try to light fires in the chamber multiple times, only for the fires to go out because they had modified the cooling loop to be too efficient. 

6: SM goes boom? Clearly you weren't an FTL13 player. Veteran FTL13 players know what a runaway TEG reactior looks like, and it ain't pretty.

Isy. I knew you'd come 'round to appreciate the diversity! And I must say even though I was pulling my hair out watching you while you set up a mad-lad fusion mix I was none the less entertained.

**Comment by ike709** [comment]
Until someone designs a new engine, this is our best bet with radiation being on its way out.

**Comment by ike709** [comment]
It'd be appreciated if you could convert Box and Meta as well @Vasily2013

**Comment by SergeiKoralev** [comment]
I can do that!
